presto-docs/src/main/sphinx/security/hive.rst

```diff
@@ -0,0 +1,237 @@
+                                                   principal specified by hive.hdfs.presto.principal.
+
+================================================== ============================================================ ==========
+
+.. _hive-security-simple:
+
+``SIMPLE``
+^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=SIMPLE
+
+When using ``SIMPLE`` authentication, Presto accesses HDFS as the user the
+Presto process is running as. This is the default authentication type, and no
+additional properties need to be set. Kerberos is not used.
+
+.. _hive-security-kerberos:
+
+``KERBEROS``
+^^^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=KERBEROS
+    hive.hdfs.presto.principal=hdfs@EXAMPLE.COM
+    hive.hdfs.presto.keytab=/etc/presto/hdfs.keytab
+
+When using ``KERBEROS`` authentication, Presto accesses HDFS as the principal
+specified by the ``hive.hdfs.presto.principal`` property. Presto authenticates
+this principal using the keytab specified by the ``hive.hdfs.presto.keytab``
+keytab.
+
+Keytab files must be distributed to every node in the cluster.
+
+:ref:`Additional information on keytab files.<hive-security-additional-keytab>`
+
+End user impersonation
+======================
+
+Impersonation Accessing HDFS
+----------------------------
+
+Presto can impersonate the end user who is running a query. In the case of a
+user running a query from the command line interface, this is the value of the
+``--user`` option. Impersonating the end user can provide additional security
+when accessing HDFS. 
+
+.. _hive-security-simple-impersonation:
+
+``SIMPLE`` authentication with HDFS impersonation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=SIMPLE
+    hive.hdfs.impersonation=true
+
+When using ``SIMPLE`` authentication with impersonation, Presto impersonates
+the user who is running the query when accessing HDFS. The user Presto is
+running as must be allowed to impersonate this user, as discussed in the
+section :ref:`configuring-hadoop-impersonation`.
+
+.. _hive-security-kerberos-impersonation:
+
+``KERBEROS`` authentication with HDFS impersonation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=KERBEROS
+    hive.hdfs.impersonation=true
+    hive.hdfs.presto.principal=presto@EXAMPLE.COM
+    hive.hdfs.presto.keytab=/etc/presto/hdfs.keytab
+
+When using ``KERBEROS`` authentication with impersonation, Presto impersonates
+the user who is running the query when accessing HDFS. The principal
+specified by the ``hive.hdfs.presto.principal`` property must be allowed to
+impersonate this user, as discussed in the section
+:ref:`configuring-hadoop-impersonation`. Presto authenticates
+``hive.hdfs.presto.principal`` using the keytab specified by the
+``hive.hdfs.presto.keytab`` keytab.
```

Comment:
I think it would be better to say:
using the keytab specified by ``hive.hdfs.presto.keytab``.